lib/src/core/services/groups_service.dart

```diff
@@ -0,0 +1,58 @@
+import 'dart:convert';
+
+import 'package:isar/isar.dart';
+import 'package:kana_to_kanji/src/core/constants/alphabets.dart';
+import 'package:kana_to_kanji/src/core/models/group.dart';
+import 'package:kana_to_kanji/src/core/services/api_service.dart';
+import 'package:kana_to_kanji/src/locator.dart';
+import 'package:http/http.dart' as http;
+
+class GroupsService {
+  final Isar _isar = locator<Isar>();
+  final ApiService _apiService = locator<ApiService>();
+
+  /// Get all the groups related to the alphabet given in parameter.
+  Future<List<Group>> getGroups(Alphabets alphabet,
+      {bool reload = false}) async {
+    final groups = _isar.groups.where().alphabetEqualTo(alphabet).findAll();
+
+    if (reload || groups.isEmpty) {
+      deleteAll();
+      return loadCollection().then(
+          (_) => _isar.groups.where().alphabetEqualTo(alphabet).findAll());
+    }
+
+    return Future.value(groups);
+  }
+
+  /// Delete all the groups.
+  void deleteAll() {
+    _isar.write((isar) => {isar.groups.where().deleteAll()});
+  }
+
+  /// Load all the groups from the API.
+  Future<dynamic> loadCollection() {
```

Comment:
We do agree that this section is only temporary as it's only until we have the ability to check the difference between the local DB and the remote one, right?